### PR TITLE
feat!: upgrade to Cosmos SDK v47

### PR DESCRIPTION
**Pull request by jgimeno**
# Description

What does this PR do?

# Purpose

Why is this PR important?

### CODE REVIEW

File: contrib/devtools/Dockerfile
Comment (k-yang) [inline contrib/devtools/Dockerfile:1]: Should we just replace the top-level Dockerfile?

File: contrib/devtools/Makefile
Comment (k-yang) [inline contrib/devtools/Makefile:1]: Should we add this to the `contrib/make` directory?

File: contrib/scripts/protocgen2.sh
Comment (k-yang) [inline contrib/scripts/protocgen2.sh:1]: Don't we already do this in `contrib/scripts/protogen.sh`?

Comment (jgimeno) [reply contrib/scripts/protocgen2.sh:1]: done

**Review by k-yang — APPROVED** [review]

**Pull request merged**